src/cunumeric/index/select.cc

```diff
@@ -0,0 +1,77 @@
+/* Copyright 2023 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/index/select.h"
+#include "cunumeric/index/select_template.inl"
+
+namespace cunumeric {
+
+using namespace legate;
+
+template <Type::Code CODE, int DIM>
+struct SelectImplBody<VariantKind::CPU, CODE, DIM> {
+  using VAL = legate_type_of<CODE>;
+
+  void operator()(const AccessorWO<VAL, DIM>& out,
+                  const std::vector<AccessorRO<bool, DIM>>& condlist,
+                  const std::vector<AccessorRO<VAL, DIM>>& choicelist,
+                  VAL default_val,
+                  const Rect<DIM>& rect,
+                  const Pitches<DIM - 1>& pitches,
+                  bool dense) const
+  {
+    const size_t volume = rect.volume();
+    uint32_t narrays    = condlist.size();
+#ifdef DEBUG_CUNUMERIC
+    assert(narrays == choicelist.size());
+#endif
+
+    if (dense) {
+      auto outptr = out.ptr(rect);
+      for (size_t idx = 0; idx < volume; ++idx) outptr[idx] = default_val;
+      for (int32_t c = (narrays - 1); c >= 0; c--) {
+        auto condptr   = condlist[c].ptr(rect);
+        auto choiseptr = choicelist[c].ptr(rect);
+        for (int32_t idx = (volume - 1); idx >= 0; idx--) {
```

Comment:
The backward loop is necessary to satisfy following requirement:
```
 When multiple conditions are satisfied, the first one encountered in condlist is used.
```

consider the following use case:
```
import numpy as np

x = np.arange(36). reshape((6,6,))
condlist = (x%2).astype(bool)
choicelist = x**2
x= np.select(condlist, choicelist, 100)
print (x)
```

it will produce:
```
[100   1 100   9 100  25]
```

Reply:
yes, you are right. I have just got confused